trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/remote/PathStyleRemoteS3Facade.java

```diff
@@ -13,37 +13,37 @@
  */
 package io.trino.s3.proxy.server.remote;
 
-import com.google.common.annotations.VisibleForTesting;
 import jakarta.ws.rs.core.UriBuilder;
 
 import java.net.URI;
 import java.util.Optional;
 
+import static io.trino.s3.proxy.server.remote.AwsRemoteS3FacadeConstants.PATH_BUILDER;
 import static java.util.Objects.requireNonNull;
 
 public class PathStyleRemoteS3Facade
         implements RemoteS3Facade
 {
-    private final String domain;
+    private final RemoteS3HostBuilder hostBuilder;
     private final boolean https;
     private final Optional<Integer> port;
 
     public PathStyleRemoteS3Facade()
     {
-        this("amazonaws.com", true, Optional.empty());
+        this(PATH_BUILDER, true, Optional.empty());
```

Comment:
The idea is that something other than `AwsRemoteS3FacadeConstants` could be used? Maybe we can generalize an "AWS" remote that binds all the generic instances with AWS specific constants instead of hard coding?